index.js

```diff
@@ -1,5 +1,62 @@
 import ManyKeysMap from 'many-keys-map';
 import pDefer from 'p-defer';
+import Queue from 'yocto-queue';
+
+function createAsyncIterator() {
+	const queue = new Queue();
+	let isComplete = false;
+	const onNextCallbacks = new Set();
```

Comment:
I think all this should be in the iterator so it can be used multiple times.

Reply:
If an iterator is created for a second time, should it replay the values that the first one gave?

Reply:
No